backend/typescript/services/implementations/storageService.ts

```diff
@@ -0,0 +1,79 @@
+import { storage } from "firebase-admin";
+import IFileStorageService from "../interfaces/storageService";
+import logger from "../../utilities/logger";
+
+const Logger = logger(__filename);
+class FileStorageService implements IFileStorageService {
+  bucket: any;
+
+  constructor() {
+    this.bucket = storage().bucket(
+      "gs://uw-blueprint-starter-code.appspot.com",
+    );
+  }
+
+  async getFile(
+    fileName: string,
+    expirationTimeMinutes: number = 60,
+  ): Promise<string> {
+    const expirationDate = new Date();
+    expirationDate.setMinutes(
+      expirationDate.getMinutes() + expirationTimeMinutes,
+    );
+    try {
+      const res = await this.bucket.file(fileName).getSignedUrl({
+        action: "read",
+        expires: expirationDate,
+      });
+      return res[0];
+    } catch (error) {
+      Logger.error(`Failed to get entity. Reason = ${error.message}`);
+      throw error;
+    }
+  }
+
+  async createFile(fileName: string, filePath: string): Promise<void> {
+    try {
+      const currentBlob = await this.bucket.file(fileName);
+      if ((await currentBlob.exists())[0]) {
+        throw new Error(`File name ${fileName} already exists`);
+      }
+      await this.bucket.upload(filePath, {
+        destination: fileName,
+      });
```

Comment:
I think you'll need to indicate content type through the `metadata` field

ref: https://googleapis.dev/nodejs/storage/latest/Bucket.html#upload, https://cloud.google.com/storage/docs/json_api/v1/objects/insert#request_properties_JSON

Probably this, check the syntax though
```suggestion
      await this.bucket.upload(filePath, {
        destination: fileName,
        metadata: { contentType }
      });
```